python/ray/serve/tests/test_task_processor.py

```diff
@@ -0,0 +1,294 @@
+
+        @serve.deployment
+        @task_consumer(task_processor_config=processor_config)
+        class ServeTaskConsumer:
+            def __init__(self):
+                self.data_received = None
+                self.task_received = False
+
+            @task_handler(name="process_request")
+            def process_request(self, data):
+                self.task_received = True
+                self.data_received = data
+                raise ValueError("Task failed as expected")
+
+        serve.run(ServeTaskConsumer.bind())
+        task_id_ref = send_request_to_queue.remote("test_data_1")
+        task_id = ray.get(task_id_ref)
+
+        def assert_result():
+            status_ref = get_task_status.remote(task_id)
+            status = ray.get(status_ref)
+
+            if (
+                status["status"] == "FAILURE"
+                and status["result"] is not None
+                and isinstance(status["result"], ValueError)
+                and str(status["result"]) == "Task failed as expected"
+            ):
+                return True
+            else:
+                return False
+
+        wait_for_condition(assert_result, timeout=5)
+
+    def test_task_consumer_as_serve_deployment_with_async_task_handler(
+        self, temp_queue_directory, serve_instance
+    ):
+        """Test that task consumers can be used as Ray Serve deployments with async task handlers."""
+
+        backend_url = "rpc://"
+        broker_url = "filesystem://"
+        queue_name = "my_default_app_queue"
+        data_folder_queue = temp_queue_directory["queue_path"]
+        control_path = temp_queue_directory["control_path"]
+
+        transport_options = {
+            "data_folder_in": data_folder_queue,
+            "data_folder_out": data_folder_queue,
+            "data_folder_processed": data_folder_queue,
+            "control_folder": control_path,
+        }
+
+        processor_config = TaskProcessorConfig(
+            queue_name=queue_name,
+            adapter_config=CeleryTaskProcessorConfig(
+                broker_url=broker_url,
+                backend_url=backend_url,
+                broker_transport_options=transport_options,
+            ),
+        )
+
+        @ray.remote(num_cpus=2)
+        def send_request_to_queue(data):
+            celery_adapter = get_task_adapter(config=processor_config)
+            result = asyncio.run(
+                celery_adapter.enqueue_task("process_request", args=[data])
+            )
+
+            assert result.id is not None
+            assert result.status in ("PENDING", "FAILED", "SUCCESS")
+
+        @serve.deployment
+        @task_consumer(task_processor_config=processor_config)
+        class ServeTaskConsumer:
+            def __init__(self):
+                self.data_received = None
+                self.task_received = False
+
+            # async task handler is not supported yet, so we should not receive the task
```

Comment:
I don't think it should silently let it go if the user used an async def for a celery task handler. Let's raise an exception in the framework?